cpp/src/parquet/file_reader.cc

```diff
@@ -366,9 +367,13 @@ class SerializedFile : public ParquetFileReader::Contents {
     std::vector<::arrow::io::ReadRange> ranges;
     prebuffered_column_chunks_.clear();
     for (int row : row_groups) {
-      std::unordered_set<int>& prebuffered = prebuffered_column_chunks_[row];
+      std::shared_ptr<Buffer>& col_bitmap = prebuffered_column_chunks_[row];
+      int num_cols = file_metadata_->num_columns();
+      PARQUET_THROW_NOT_OK(
+          AllocateBitmap(num_cols, properties_.memory_pool()).Value(&col_bitmap));
+      ::arrow::bit_util::ClearBitmap(col_bitmap->mutable_data(), 0, num_cols);
```

Comment:
Should we just call `::memset(col_bitmap->mutable_data(), 0, col_bitmap->size())` ? It would be less efficient to call `ClearBitmap` if `num_cols` is not a multiply of 8.